### PR TITLE
Address APIView comments for Communication

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/communication/Azure.ResourceManager.Communication/api/Azure.ResourceManager.Communication.netstandard2.0.cs [hunk 1]
```diff
@@ -171,40 +171,47 @@ public partial class EmailServiceResourceData : Azure.ResourceManager.Models.Tra
...
+        public static bool operator !=(Azure.ResourceManager.Communication.Models.CommunicationNameAvailabilityReason left, Azure.ResourceManager.Communication.Models.CommunicationNameAvailabilityReason right) { throw null; }
         public override string ToString() { throw null; }
     }
-    public partial class CheckNameAvailabilityRequestBody
+    public partial class CommunicationNameAvailabilityRequestContent
     {
-        public CheckNameAvailabilityRequestBody() { }
+        public CommunicationNameAvailabilityRequestContent() { }
         public string Name { get { throw null; } set { } }
         public string ResourceType { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/communication/Azure.ResourceManager.Communication/api/Azure.ResourceManager.Communication.netstandard2.0.cs:207]: This is not changed as the original property name was `type` in swagger.

File: sdk/communication/Azure.ResourceManager.Communication/api/Azure.ResourceManager.Communication.netstandard2.0.cs [hunk 2]
```diff
@@ -171,40 +171,47 @@ public partial class EmailServiceResourceData : Azure.ResourceManager.Models.Tra
...
     }
+    public partial class CommunicationDomainResourcePatch : Azure.ResourceManager.Communication.Models.CommunicationAcceptTags
+    {
+        public CommunicationDomainResourcePatch() { }
+        public Azure.ResourceManager.Communication.Models.UserEngagementTracking? UserEngagementTracking { get { throw null; } set { } }
+        public System.Collections.Generic.IDictionary<string, string> ValidSenderUsernames { get { throw null; } }
+    }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
-    public readonly partial struct CheckNameAvailabilityReason : System.IEquatable<Azure.ResourceManager.Communication.Models.CheckNameAvailabilityReason>
+    public readonly partial struct CommunicationNameAvailabilityReason : System.IEquatable<Azure.ResourceManager.Communication.Models.CommunicationNameAvailabilityReason>
```
Comment (ArcturusZhang) [inline sdk/communication/Azure.ResourceManager.Communication/api/Azure.ResourceManager.Communication.netstandard2.0.cs:186]: Should this be better if it is called `CommunicationNameUnvailableReason`?

File: sdk/communication/Azure.ResourceManager.Communication/api/Azure.ResourceManager.Communication.netstandard2.0.cs [hunk 3]
```diff
@@ -171,40 +171,47 @@ public partial class EmailServiceResourceData : Azure.ResourceManager.Models.Tra
...
-        public static implicit operator Azure.ResourceManager.Communication.Models.CheckNameAvailabilityReason (string value) { throw null; }
-        public static bool operator !=(Azure.ResourceManager.Communication.Models.CheckNameAvailabilityReason left, Azure.ResourceManager.Communication.Models.CheckNameAvailabilityReason right) { throw null; }
+        public static bool operator ==(Azure.ResourceManager.Communication.Models.CommunicationNameAvailabilityReason left, Azure.ResourceManager.Communication.Models.CommunicationNameAvailabilityReason right) { throw null; }
+        public static implicit operator Azure.ResourceManager.Communication.Models.CommunicationNameAvailabilityReason (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.Communication.Models.CommunicationNameAvailabilityReason left, Azure.ResourceManager.Communication.Models.CommunicationNameAvailabilityReason right) { throw null; }
         public override string ToString() { throw null; }
     }
-    public partial class CheckNameAvailabilityRequestBody
+    public partial class CommunicationNameAvailabilityRequestContent
```
Comment (ArcturusZhang) [inline sdk/communication/Azure.ResourceManager.Communication/api/Azure.ResourceManager.Communication.netstandard2.0.cs:203]: Maybe we should also remove the Request here?
Like this: https://github.com/Azure/azure-sdk-for-net/blob/7f9ce3397d1aa193c9f9e6b2ea6c61f4e51b6c08/sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/ManagementGroupNameAvailabilityContent.cs#L13

Comment (ArthurMa1978) [reply sdk/communication/Azure.ResourceManager.Communication/api/Azure.ResourceManager.Communication.netstandard2.0.cs:203]: Make sense, fixed

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**